### PR TITLE
Closes QOM when file or project search opens

### PR DESCRIPTION
**Pull request by lukaszsobek**
Associated Issue: #5416

Closes quick open modal when either file search or project search are being opened.

Vid:
closes-qom

### CODE REVIEW

File: src/components/Editor/SearchBar.js [hunk 1]
```diff
@@ -148,13 +149,14 @@ class SearchBar extends Component<Props, State> {
...
     e.stopPropagation();
     e.preventDefault();
-    const { editor } = this.props;
+    const { editor, closeQuickOpen, searchOn, setActiveSearch } = this.props;
 
-    if (!this.props.searchOn) {
-      this.props.setActiveSearch("file");
+    if (!searchOn) {
+      closeQuickOpen();
+      setActiveSearch("file");
```
Comment (jasonLaster) [inline src/components/Editor/SearchBar.js]: can we have `setActiveSearch` close quick open if it's open? that will simplify the component

File: src/components/ProjectSearch.js [hunk 2]
```diff
@@ -117,14 +118,16 @@ export class ProjectSearch extends Component<Props, State> {
...
+    const { closeProjectSearch, setActiveSearch, closeQuickOpen } = this.props;
     if (e) {
       e.preventDefault();
     }
 
     if (this.isProjectSearchEnabled()) {
       return closeProjectSearch();
     }
+
+    closeQuickOpen();
```
Comment (jasonLaster) [inline src/components/ProjectSearch.js]: same here

File: src/components/Editor/SearchBar.js (hunk 1 above)
Comment (lukaszsobek) [reply src/components/Editor/SearchBar.js]: Sure.

File: src/actions/ui.js [hunk 3]
```diff
@@ -38,6 +38,8 @@ export function setActiveSearch(activeSearch?: ActiveSearchType) {
       return;
     }
 
+    dispatch({ type: "CLOSE_QUICK_OPEN" });
```
Comment (jasonLaster) [inline src/actions/ui.js]: lets add an if statement so we only fire it if it's open.

we can also add a unit test here where we open the modal and then fire SAS

**Comment by jasonLaster** [comment]
try rebasing

**Comment by darkwing** [comment]
I'm still seeing layering with the PR:

dialogs

I did CMD+F, then CMD+P, and they both display.

**Comment by lukaszsobek** [comment]
@darkwing - sure, from what I understood it was supposed to close the modal when we open projectsearch or searchbar, not prevent opening it altogether. So:
- when the modal `cmd+p` is open and you open `cmd+shift+f` or `cmd+f` the modal will close.
- when `cmd+shift+f` or `cmd+f` is already open and you press `cmd+p` it will not prevent opening the modal since you clearly intended to open the modal.

We might be talking about another issue here: "if projectsearch/searchbar is open and we open the modal, should we close the projectsearch/searchbar?". The issue is very much related, but I see two things here:
1.) do we want that behavior?
2.) should this issue be handled separately as it seems to refer to a different part of UI behavior?

**Pull request merged**